### ISSUE TITLE
Additional Docker-Compose Examples

### ISSUE THREAD

**Issue by ammacdonald3** [will-moss/erin#4]
Thanks so much for creating this! Do you by chance have a simple docker-compose example to run this on a headless server exclusively on a LAN? I'm having trouble understanding the concept of a PUBLIC_URL for an internal app. Additionally, I'm already using ports 80 and 443 for other services, so I'd like to remap to other ports.

The below docker-compose starts the container on my server, but I still can't connect via web browser from my laptop. Do you spot anything I should change? 

```yaml
services:
  erin:
    image: mosswill/erin:latest
    ports:
      - "447:443"
    volumes:
      - ./videos:/srv/videos:ro
    environment:
      PUBLIC_URL: "https://192.168.0.200:447"
      AUTH_ENABLED: "false"

```

Below are the logs from the erin-erin-1 container:

```
INF ts=1724121604.5639763 msg=using config from file file=/etc/caddy/Caddyfile

INF ts=1724121604.5666578 msg=adapted config to JSON adapter=caddyfile

WRN ts=1724121604.5681186 msg=Caddyfile input is not formatted; run 'caddy fmt --overwrite' to fix inconsistencies adapter=caddyfile file=/etc/caddy/Caddyfile line=11

INF ts=1724121604.5696566 logger=admin msg=admin endpoint started address=localhost:2019 enforce_origin=false origins=["//localhost:2019","//[::1]:2019","//127.0.0.1:2019"]

INF ts=1724121604.5721638 logger=http.auto_https msg=enabling automatic HTTP->HTTPS redirects server_name=srv0

INF ts=1724121604.572248 logger=tls.cache.maintenance msg=started background certificate maintenance cache=0xc0005e0e80

INF ts=1724121604.590767 logger=http msg=enabling HTTP/3 listener addr=:447

INF ts=1724121604.591296 msg=failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 7168 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes for details.

INF ts=1724121604.5930288 logger=http.log msg=server running name=srv0 protocols=["h1","h2","h3"]

INF ts=1724121604.5933332 logger=http.log msg=server running name=remaining_auto_https_redirects protocols=["h1","h2","h3"]

INF ts=1724121604.5933745 logger=http msg=enabling automatic TLS certificate management domains=["192.168.0.200"]

WRN ts=1724121604.5935032 logger=pki.ca.local msg=installing root certificate (you might be prompted for password) path=storage:pki/authorities/local/root.crt

INF ts=1724121604.593768 msg=define JAVA_HOME environment variable to use the Java trust

INF ts=1724121604.593813 msg=warning: "certutil" is not available, install "certutil" with "apt install libnss3-tools" or "yum install nss-tools" and try again

INF ts=1724121604.5937781 logger=tls.obtain msg=acquiring lock identifier=192.168.0.200

INF ts=1724121604.6061194 logger=tls msg=cleaning storage unit storage=FileStorage:/data/caddy

INF ts=1724121604.606681 logger=tls msg=finished cleaning storage units

INF ts=1724121604.6087298 logger=tls.obtain msg=lock acquired identifier=192.168.0.200

INF ts=1724121604.6088006 logger=tls.obtain msg=obtaining certificate identifier=192.168.0.200

INF ts=1724121604.6099973 logger=tls.obtain msg=certificate obtained successfully identifier=192.168.0.200 issuer=local

INF ts=1724121604.6100614 logger=tls.obtain msg=releasing lock identifier=192.168.0.200

WRN ts=1724121604.6103618 logger=tls msg=stapling OCSP error=no OCSP stapling for [192.168.0.200]: no OCSP server specified in certificate identifiers=["192.168.0.200"]

INF ts=1724121604.6410282 msg=certificate installed properly in linux trusts

INF ts=1724121604.641462 msg=autosaved config (load with --resume flag) file=/config/caddy/autosave.json

INF ts=1724121604.6415918 msg=serving initial configuration
```

**Comment by will-moss** [comment]
Hi,

Thanks a lot for the kind words, and for all those details.

I think the problem in your configuration is the following : 
- You have set `PUBLIC_URL` to : `XXX:447` : So, your server, inside the Docker container, will listen on the port 447.
- You have set a port mapping equal to `447:443` : It means that you expect to connect (from your laptop) on port 447, while Docker will redirect the request to the container on the port 443.
- However, your container is actually listening on 447, so Docker sends the request to nowhere (443).

I think that a quick way to assess my assumption would be to, firstly, change your port mapping from `447:443` to `447:447`, so the requests transit properly from your laptop to your remote Docker container. Also, you must make sure that the port 447 is open on your server, and not blocked by any firewall. If you are on Debian or alike, the command `ufw` can help, with `ufw allow 447` for example, to open that port.

Regarding `PUBLIC_URL`, here's a link to the official Caddy documentation, showing all the formats you can use : https://caddyserver.com/docs/caddyfile/concepts#addresses (You may want to read the `Automatic HTTPS` part thoroughly if you encounter any issues with that. The goal of the `PUBLIC_URL` variable is to configure Caddy (the server that powers everything in the app) to let it know : What port should we listen on? What domain / IP should we reply to? Should be automatically generate an HTTPS certificate? etc.

Does it help? Let me know!

**Comment by ammacdonald3** [comment]
Thanks for the quick reply! You were correct that I should change my port mapping to `447:447`. I also needed to change
`PUBLIC_URL: "https://192.168.0.200:447"` to `PUBLIC_URL: "http://192.168.0.200:447"` (i.e. https to http). This is my preferred approach, as I already use Nginx Proxy Manager on my network to secure self-hosted apps with a domain name. I'm not sure if it's worth calling this out in the documentation, but I've included the full docker-compose file below if it helps. Thanks again for building this great product!

```
services:
  erin:
    image: mosswill/erin:latest
    ports:
      - "447:447"
    volumes:
      - ./videos:/srv/videos:ro
    environment:
      PUBLIC_URL: "http://192.168.0.200:447"
      AUTH_ENABLED: "false"
```
